### PR TITLE
feat(metadata): add general server payload maximum to secret scan preferences

### PR DESCRIPTION
**Pull request by fnareoh**
I hesitated on where to store the server payload in the GCC client but as I want to use it for secret scanning it seemed best there.
I also added a default for the payload if it isn't sent in preference, set to 25Mb, possibly to discuss.

### CODE REVIEW

File: pygitguardian/client.py [hunk 1]
```diff
@@ -584,7 +584,10 @@ def read_metadata(self) -> Optional[Detail]:
             result.status_code = resp.status_code
             return result
         metadata = ServerMetadata.from_dict(resp.json())
-
+        if "general__maximum_payload_size" in metadata.preferences:
+            metadata.secret_scan_preferences.general__maximum_payload_size = (
+                metadata.preferences["general__maximum_payload_size"]
+            )
```
Comment (agateau-gg) [inline pygitguardian/client.py:590]: I think the maximum payload size should be a property of GGClient: all verticals could need it. I would turn this into:

```python
self.maximum_payload_size = metadata.preferences.get("general__maximum_payload_size", MAXIMUM_PAYLOAD_SIZE)
```

File: pygitguardian/config.py [hunk 2]
```diff
@@ -4,3 +4,4 @@
 
 MULTI_DOCUMENT_LIMIT = 20
 DOCUMENT_SIZE_THRESHOLD_BYTES = 1048576  # 1MB
+GENERAL_MAXIMUM_PAYLOAD = 2621440  # 25MB
```
Comment (agateau-gg) [inline pygitguardian/config.py:7]: The `general__` part of the field name is an implementation detail of the server. I'd rather hide it and name this constant `MAXIMUM_PAYLOAD_SIZE`.

**Review by agateau-gg — CHANGES_REQUESTED** [review]

**Review by agateau-gg — CHANGES_REQUESTED** [review]
Looks good, just needs a changelog entry to document the new field. See https://github.com/GitGuardian/ggshield/blob/main/doc/dev/getting-started.md#changelog for details.

**Comment by fnareoh** [comment]
Should be good now ? Sorry for missing scriv and thank you for the patient reviews :)

**Comment by agateau-gg** [comment]
> GGShield fails in the lint job fails because the PR comes from a fork. I filed #87 to avoid that. The job should become green once this change is merged.

Ah, that did not go as expected: looks like GitHub takes the CI config from the branch being merged instead of the result of the merge. Force-merging it.

**Pull request merged**